src/rules/no-import-module-exports.js

```diff
@@ -0,0 +1,72 @@
+import minimatch from 'minimatch'
+import path from 'path'
+import readPkgUp from 'read-pkg-up'
+
+function getEntryPoint(context) {
+  try {
```

Comment:
The `require.resolve` approach seems to work well. The only thing to take into consideration is the use of `read-pkg-up` for finding the path to the nearest `package.json`. Using `read-pkg-up` the implementation would look something like:
```js
import path from 'path'
import readPkgUp from 'read-pkg-up'

function getEntryPoint(context) {
    const pkg = readPkgUp.sync({ cwd: context.getFilename() })
    return require.resolve(path.dirname(pkg.path))
}
```

But since all we need is the path to the nearest `package.json` (we don't actually need to parse it) we could use the [`pkg-up`](https://www.npmjs.com/package/pkg-up) lib instead. In which case the implementation would look like:

```js
import path from 'path'
import pkgUp from 'pkg-up'

function getEntryPoint(context) {
  const pkgPath = pkgUp.sync(context.getFilename())
  return require.resolve(path.dirname(pkgPath))
}
```

So, we can:
A) Stick to the current implementation where we parse the `package.json`, and create the path to "main" ourselves.
B) Use `read-pkg-up` to find the path to the nearest `package.json`, then use `require.resolve` to get the path to `main`.
C) Install and use `pkg-up` to find the path to the nearest `package.json`, then use `require.resolve` to get the path to `main`.

I prefer B and C over A....and lean towards C, but I'm not sure what the cost would be of adding another dependency. What do you think?